src/tools/crossgen2/Common/TypeSystem/Common/MethodDesc.cs

```diff
@@ -237,7 +271,8 @@ public MethodSignature ToSignature()
                 _returnType != _template._returnType ||
                 _parameters != _template._parameters)
             {
-                _template = new MethodSignature(_flags, _genericParameterCount, _returnType, _parameters);
+                // TODO - This is sketchy, are we sure keeping the custom modifier unchanged is good?
```

Comment:
This isn't ok when the MethodSignatureBuilder is used as part of Instantiating a sig for the purpose of a derived type overriding a method. For example, in C# like pseudocode the Derived.Function method needs to match with the instantiated Base<T,U>.Function method.

```
class Base<U,T>
{
    public abstract void Function(Dictionary<U, cmod T>);
}

class Derived : Base<List<int>, string>
{
    public override void Function(Dictionary<List<int>, cmod string>) {} 
}
```

Reply:
The new version should work with this scenario, need testing.

Reply:
I manually ported the changes to CoreRT and used this new unit test to verify that matching instantiated signature will work.

https://github.com/dotnet/corert/pull/7869